src/features/canvass/components/LocationDialog/pages/HouseholdVisitPage.tsx

```diff
@@ -105,28 +106,56 @@ const HouseholdVisitPage: FC<HouseholdVisitPageProps> = ({
           return (
             <Step key={index}>
               <StepButton
-                onClick={() => setStep(index)}
+                onClick={() => {
+                  if (index < step) {
+                    const newResponses: Record<number, MetricResponse> = {};
+                    metrics.forEach((m, i) => {
+                      if (i < index && responseByMetricId[m.id]) {
+                        newResponses[m.id] = responseByMetricId[m.id];
+                      }
+                    });
+                    setResponseByMetricId(newResponses);
+                  }
+
+                  setStep(index);
+                }}
                 sx={{
+                  '& .MuiStepLabel-vertical': {
+                    alignItems: 'start',
+                  },
                   '& span': {
                     overflow: 'hidden',
                   },
+                  display: 'block',
                 }}
               >
-                <Typography
-                  sx={{
-                    overflow: 'hidden',
-                    textOverflow: 'ellipsis',
-                    whiteSpace: stepIsCurrent ? 'normal' : 'nowrap',
-                  }}
-                >
-                  {metric.question}
-                </Typography>
+                <Box sx={{ width: '100%' }}>
+                  <Box
+                    sx={{
+                      alignItems: 'center',
+                      display: 'flex',
+                      justifyContent: 'space-between',
+                      width: '100%',
+                    }}
+                  >
+                    <Typography
+                      sx={{
+                        overflow: 'hidden',
+                        textOverflow: 'ellipsis',
+                        whiteSpace: stepIsCurrent ? 'normal' : 'nowrap',
+                      }}
+                    >
+                      {metric.question}
+                    </Typography>
+                    {!stepIsCurrent && completed && <Undo />}
```

Comment:
I don't think `completed` is the right condition here, because it means that there is no "Undo" button on skipped questions (which are not considered completed). Maybe use the index instead, i.e. whether `index < step`?